cmd/horcrux/cmd/leader_election_test.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLeaderElectionMultiAddressDomain(t *testing.T) {
+	cfg := &CosignerConfig{
+		P2PListen: "tcp://signer-1:2222",
+		Peers: []CosignerPeer{
+			{
+				P2PAddr: "tcp://signer-2:2222",
+			},
+			{
+				P2PAddr: "tcp://signer-3:2222",
+			},
+		},
+	}
+
+	multiAddress := leaderElectMultiAddress(cfg)
+
+	require.Equal(t, "multi:///signer-1:2222,signer-2:2222,signer-3:2222", multiAddress)
+}
+
+func TestLeaderElectionMultiAddressIPv4(t *testing.T) {
+	cfg := &CosignerConfig{
+		P2PListen: "tcp://10.0.0.1:2222",
+		Peers: []CosignerPeer{
+			{
+				P2PAddr: "tcp://10.0.0.2:2222",
+			},
+			{
+				P2PAddr: "tcp://10.0.0.3:2222",
+			},
+		},
+	}
+
+	multiAddress := leaderElectMultiAddress(cfg)
```

Comment:
This is opinion.

Testing private functions is generally considered bad practice. See [1](https://softwareengineering.stackexchange.com/questions/380287/why-is-unit-testing-private-methods-considered-as-bad-practice), [2](https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html), [3](https://medium.com/@thrawn01/why-you-should-never-test-private-methods-f822358e010).

I never test private (unexported) methods/functions. Private functionality should be able to change with impunity. A public function clearly signals to the developer that some part of the codebase depends on it, so we have to take greater care of refactoring it. Private functions typically don't signal that importance. 

Additionally, a package with public functions that `main()` functions import surfaces the overall design and components of the system. Vs. hiding them in `main()`.

In this case, I'd recommend creating a new package outside of `cmd` that may have something to do with `leader`.

Reply:
good catch, moved these address utils to a `client` package